src/otx/engine/engine.py

```diff
@@ -598,6 +605,26 @@ def from_config(cls, config_path: PathLike, data_root: PathLike | None = None, *
     # Property and setter functions provided by Engine.
     # ------------------------------------------------------------------------ #
 
+    @property
+    def work_dir(self) -> PathLike:
+        """Work directory."""
+        return self._work_dir
+
+    @work_dir.setter
+    def work_dir(self, val: PathLike) -> None:
```

Comment:
An argument named `val` seems odd. Could you rename it?
The same goes for device setter functions.

Reply:
done at https://github.com/openvinotoolkit/training_extensions/pull/2912/commits/f209dd6e9b7127a4ea0353c8d8c0b485baa6ef6f